lua/neogit/status.lua

```diff
@@ -289,18 +306,26 @@ local function draw_buffer()
 
       for _, f in ipairs(data.items) do
         local label = util.pad_right(format_mode(f.mode), max_len)
+
         if label and vim.o.columns < 120 then
           label = vim.trim(label)
         end
 
+        local line
         if f.mode and f.original_name then
-          output:append(string.format("%s %s -> %s", label, f.original_name, f.name))
+          line = string.format("%s %s -> %s", label, f.original_name, f.name)
         elseif f.mode then
-          output:append(string.format("%s %s", label, f.name))
+          line = string.format("%s %s", label, f.name)
         else
-          output:append(f.name)
+          line = f.name
+        end
+
+        if f.submodule then
+          line = line .. " " .. format_submodule_mode(f.submodule)
```

Comment:
Definitely `string.format` here. Lua performance can go to shit with a lot of string concatenation.